src/App.tsx

```diff
@@ -1,27 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import {
+  get5First,
+  getAll,
+  getRedGoods,
+} from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [displayedGoods, setDisplayedGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const loadAll = async (goods: Promise<Good[]>) => {
+    setDisplayedGoods(await goods);
+  };
```

Comment:
- should be named properly
- passing a callback function that returns a promise allows you to control when the promise is executed, and gives you more flexibility to handle errors and other conditions
```js
  const handleGetGoods = async (goods: () => Promise<Good[]>) => {
  ...
```

Reply:
also, don't forget to handle errors and show appropriate content

Reply:
hopefully i've done it correctly